vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SetContextSteps.java

```diff
@@ -56,7 +60,59 @@ public void switchToNativeContext()
      * @param index index of web view
      */
     @When("I switch to web view with index `$index`")
-    public void switchToWebView(int index)
+    public void switchToWebViewByIndex(int index)
+    {
+        performOnWebViews(webViews ->
+        {
+            int webViewIndex = index - 1;
+            int size = webViews.size();
+            if (webViewIndex >= 0 && webViewIndex < size)
+            {
+                String webview = webViews.get(webViewIndex);
+                LOGGER.atInfo().addArgument(webview::toString)
+                               .addArgument(index)
+                               .log("Switching to '{}' web view found by the index {}");
+                getContextAware().context(webview);
+                return;
+            }
+
+            softAssert.recordFailedAssertion(String.format("Web view with index %s does not exist", index));
+        });
+    }
+
+    /**
+     * Switches context to a web view where name matches the rule
+     *
+     * @param rule  The web view name comparison rule: "is equal to", "contains", "does not contain" or "matches"
+     * @param value The value to find the target web view name to switch to
+     */
+    @When("I switch to web view with name which $comparisonRule `$value`")
+    public void switchToWebViewByName(StringComparisonRule rule, String value)
+    {
+        performOnWebViews(webViews ->
+        {
+            Matcher<String> webViewMatcher = rule.createMatcher(value);
+            Predicate<String> webViewFilter = webViewMatcher::matches;
+
+            List<String> matchedWebViews = webViews.stream()
+                                                   .filter(webViewFilter)
+                                                   .collect(Collectors.toList());
+
+            if (matchedWebViews.size() == 1)
+            {
+                String webView = matchedWebViews.get(0);
+                LOGGER.atInfo().addArgument(webView).log("Switching to web view with the name '{}'");
+                getContextAware().context(webView);
+                return;
+            }
+
+            softAssert.recordFailedAssertion(
+                    String.format("The number of web views with name which is %s is expected to be 1, but got %d",
+                            webViewMatcher, matchedWebViews.size()));
```

Comment:
it's better to use `rule` here, instead of matcher, because matcher produces hardly readable message